### PR TITLE
Remove all uses of WABT_VECTOR from interpreter

### PR DESCRIPTION
**Pull request by binji**

### CODE REVIEW

File: src/binary-reader-interpreter.cc [hunk 1]
```diff
@@ -890,8 +868,6 @@ static Result on_elem_segment_function_index_check(uint32_t index,
     return Result::Error;
   }
 
-  table->func_indexes[ctx->table_offset++] =
```
Comment (binji) [inline src/binary-reader-interpreter.cc]: This was actually a bug that was caught by the transition -- it shouldn't be updating the table in the "check" callback.

File: src/binary-reader.cc [hunk 2]
```diff
@@ -466,6 +466,17 @@ static void write_indent(LoggingContext* ctx) {
     LOGF_NOINDENT(__VA_ARGS__); \
   } while (0)
 
+static void logging_on_error(BinaryReaderContext* context,
```
Comment (binji) [inline src/binary-reader.cc]: This was a bug, basically the reader would crash when you turn on logging and there is an error.

Comment (sbc100) [inline src/binary-reader.cc]: Hmm.. i just removed this didn't I?  Why does it crash? 

Perhaps we need more tests with logging enabled.   Can you make this a separate PR maybe?

Comment (binji) [inline src/binary-reader.cc]: Good idea.
